packages/plugin-legacy/src/index.ts

```diff
@@ -331,24 +331,30 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
         }
 
         return (chunkInfo) => {
-          let fileName =
+          const file =
             typeof fileNames === 'function' ? fileNames(chunkInfo) : fileNames
+          const fileArr = file.split('/')
+          let fileName = fileArr.pop()!
 
           if (fileName.includes('[name]')) {
             // [name]-[hash].[format] -> [name]-legacy-[hash].[format]
             fileName = fileName.replace('[name]', '[name]-legacy')
-          } else if (fileName.includes('[hash]')) {
-            // custom[hash].[format] -> [name]-legacy[hash].[format]
-            // custom-[hash].[format] -> [name]-legacy-[hash].[format]
-            // custom.[hash].[format] -> [name]-legacy.[hash].[format]
-            fileName = fileName.replace(/[.-]?\[hash\]/, '-legacy$&')
           } else {
-            // entry.js -> entry-legacy.js
-            // entry.min.js -> entry-legacy.min.js
-            fileName = fileName.replace(/(.+?)\.(.+)/, '$1-legacy.$2')
+            if (fileName.includes('[hash]') && !fileName.startsWith('[hash]')) {
```

Comment:
Would you change this `if` to `else if` so that the diff will be smaller?

Reply:
> What is the reason to only apply the conversion to the file part? I guess /foo/[name]/[hash].[ext] is valid.

Although rollup allows this use, there is no mention of this usage in all related documentation and the default values ​​of the properties, and these placeholders: [name], [hash], [ext], [extname] are strictly speaking only can be used for the final file name.

Reply:
> done